mesonbuild/dependencies/base.py

```diff
@@ -912,7 +916,7 @@ def get_compile_args(self):
 
     def get_link_args(self):
         if self.found():
-            return ['-F' + self.path, '-framework', self.name.split('.')[0]]
+            return ['-F' + self.path, '-framework', self.name.split('.')[0], '-Wl,-rpath,' + self.path]
```

Comment:
Are you sure this will work for other frameworks? Depending on their install_name it might or might not work, I think. Though adding the rpath is probably not a bit harm, I am just not sure if this is something that should be done in general, as for most frameworks this will not be necessary, as far as I know. (So far the only frameworks relying on rpath to be set correctly that I've seen are the Qt ones)

Reply:
maybe to make it general the rpath should only be set if `self.path` is not in the default search path, but I can't find a standard macos search path (on my mac `DYLD_FRAMEWORK_PATH` is empty).

Reply:
Even then I don't think this will work in all cases, while the framework can have something like `@rpath/QtCore.framework/Versions/5/QtCore`, this could in theory be different, like: `@rpath/Frameworks/Foo.framework/Versions/1/Foo` and then your approach would not work fine anymore. In my opinion this is really something specific to a framework and should be set manually. If we know Qt requires it, maybe this can be added to the Qt module instead?

Reply:
Alternatively with some path processing and by reading the frameworks install_name, this could probably be made to work generically for all frameworks:

Psuedo-code of an approach that might work:

```
install_name = read_install_name(framework_path)

if install_name starts with '@rpath' do:
    rpath = left_remove('@rpath', install_name)
    final_rpath = right_remove(framework_path, rpath)
    # Use final_rpath
else:
    # This is not a framework using rpaths, do nothing.
```